### ISSUE TITLE
Troubleshooting a cheap breakout board

### ISSUE THREAD

**Issue by mbouy974** [SV-Zanshin/MAX31855#8]
This code nicely provides a fault code to help with troubleshooting, but the wiki didn't mention my issue: Fault code 7. I now realise this must be fault codes 1, 2 & 4 combined. It might be helpful to present the codes as they are received, 3 separate binary digits presenting different issues.

From the data sheet:  Bits D0, D1, and D2 of the output data are normally low. 
Bit D2 goes high to indicate a thermocouple short to VCC 
Bit D1 goes high to indicate a thermocouple short to GND
Bit D0 goes high to indicate a thermocouple open circuit. 

Let's put Fault code 7 aside, it means nothing was correct. What I've discovered is the breakout board I'm using actually has a track between T- and GND, which presents as fault code 2. If I remove the thermocouple I see fault code 3 because faults 1 and 2 are active.

It seems clear the breakout board is garbage and only redeemable by cutting that track if I can get a scalpel into it. However...

The datasheet also suggests T- is tied to ground when measuring the thermocouple, so why is it considered a fault condition?  I'm wondering if the board should work if I ignore errors (it doesn't), or if there's an error in the datasheet.

"When executing the temperature conversion of the external thermocouple, the connections to the internal fault-detection circuit are opened (switches S1 and S2 in the Block Diagram) and the switch connecting the coldjunction compensation circuit is opened (switch S5). The internal ground reference connection (switch S3) and the connection to the ADC (switch S4) are closed. This
allows the ADC to process the voltage detected across the T+ and T- terminals."

and later: "The fault-detection circuit tests for shorted connections to VCC or GND on the T+ and T- inputs".

![Block diagram](https://user-images.githubusercontent.com/75195302/100540532-b9910a80-328d-11eb-9544-89188c4b0d52.JPG)

I cut the track and resolved the fault condition, now I just get nonsense data. Internal temp 128 (or -128) and probe -1100 sitting on my desk.  Probe goes to -2048 in ice water and +1000 when I warm it again with my hand. It's possible the chip was binned as faulty in the first place. I've got 25 more to test.

![WIN_20201129_22_20_30_Pro](https://user-images.githubusercontent.com/75195302/100541570-b77e7a00-3294-11eb-8feb-6b5570342597.jpg)

**Comment by SV-Zanshin** [comment]
I've amended the wiki page to reflect that the error codes are additive.

I believe that the datasheet is correct and should not be regarded as suspect. 

Can you check to see whether or not the thermocouple leads are shorted to either VCC or GND using a multimeter?

**Comment by mbouy974** [comment]
No longer a short anywhere.

**Comment by SV-Zanshin** [comment]
What error code do you get if you don't attach the thermocouple? It should be 1

**Comment by mbouy974** [comment]
Yes that works as expected.

Thanks for working with me on this, but it's after midnight here and I'm
off to sleep. I'll be back on the case in about 10 hours.

**Comment by SV-Zanshin** [comment]
No worries - if there are any questions or if you've solved the issue just post it here!

Are you in Oz?

**Comment by mbouy974** [comment]
Yes, I am in South Australia. 

Apparently there are known defective MAX31855K chips in the marketplace, especially the cheap outlets like AliExpress. 

This post https://forum.arduino.cc/index.php?topic=526439.0  
links to the Russian forum:
https://radiokot.ru/forum/viewtopic.php?f=2&t=79526&sid=748afea679c3e4474e8ab02522e1c1e3&start=20

..Which names a bad batch, and I'm holding that exact batch. This is in addition to a breakout board designed to fail - they were never going to work as they were produced, even with good chips. I'll do my best to get my money back and shame the seller into withdrawing the product if I can.

I paid about AU$3 each while the equivalent from digikey is AU$24. I also waited about 6 weeks for slow delivery from China, and I know Digikey will deliver in 4 days.

My final test was with a board where I had cut the bad track, plugged direct into DIO pins and used software SPI. The intent was to remove the wiring and any noise that may introduce. It made no difference - the board just produces nonsense values, formatted correctly. The probe value does increase if warmed and decrease if cooled, and maybe that element of the chip is OK and it's the ambient/cold junction compensation which has failed. I don't believe this can be bypassed if for example I just wanted warm/not warm values with no compensation - but if anyone can think of a way, I'd be interested.

I also provided some feedback to Adafruit since their driver just produces a 'NAN' for any fault code, when it could actually be informative.
